File: Algorithm_LevelOne/Single_Number/Single_Number.py
# 136. Single Number Python

class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        dic = {}
        for i in nums:
            if i in dic:
                dic[i] += 1
            else: dic[i] = 1
        
        for i in dic:
            if dic[i] == 1:
                return i
        

File: Algorithm_LevelOne/Single_Number/Single_Number.cc
// 136. Single Number C++
#include <vector>
#include <unordered_map>
using namespace std;

class Solution {
public:
    int singleNumber(vector<int>& nums) {
        unordered_map<int, int> mp;
        for (auto x: nums) mp[x]++;
        int ans = 0;
        for (auto x: mp) {
            if (x.second == 1) {
                ans = x.first;
                break;
            }
        }
        return ans;
    }
};
